From ad66d744e2cd4a94e68967aa0128ac110b5370b5 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:19 -0400
Subject: [tools] perf/top: Add top.children config option

Message-id: <1410181752-25631-255-git-send-email-jolsa@redhat.com>
Patchwork-id: 91998
O-Subject: [PATCH RHEL7.1 BZ1134356 254/307] perf top: Add top.children config option
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 104ac991bd821773cba6f262f97a4a752ed76dd5
Author: Namhyung Kim <namhyung@kernel.org>
Date: Tue Jan 22 18:09:46 2013 +0900

description
===========
Add top.children config option for setting default value of
callchain accumulation.  It affects the output only if one of
-g or --call-graph option is given as well.

A user can write .perfconfig file like below to enable accumulation
by default:

  $ cat ~/.perfconfig
  [top]
  children = true

And it can be disabled through command line:

  $ perf top --no-children
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-top.c b/tools/perf/builtin-top.c
index fea55e3..377971d 100644
--- a/tools/perf/builtin-top.c
+++ b/tools/perf/builtin-top.c
@@ -1004,6 +1004,10 @@ static int perf_top_config(const char *var, const char *value, void *cb)
 
  if (!strcmp(var, "top.call-graph"))
   return record_parse_callchain(value, &top->record_opts);
+ if (!strcmp(var, "top.children")) {
+  symbol_conf.cumulate_callchain = perf_config_bool(var, value);
+  return 0;
+ }
 
  return perf_default_config(var, value, cb);
 }
-- 
1.7.1